File: trading-service/dto/OrderTransactionInitiationDTO.go
package dto

type OrderTransactionInitiationDTO struct {
	Uid             string  `json:"uid"`
	SellerAccountId uint    `json:"sellerAccountId"`
	BuyerAccountId  uint    `json:"buyerAccountId"`
	Amount          float64 `json:"amount"`
	Fee             float64 `json:"fee"`
	Direction       string  `json:"direction"`
}

File: banking-service/src/main/java/com/banka1/banking/dto/OrderTransactionInitiationDTO.java
package com.banka1.banking.dto;

import lombok.Data;

@Data
public class OrderTransactionInitiationDTO {
    private String uid;
    private Long sellerAccountId;
    private Long buyerAccountId;
    private Double amount;
    private Double fee;
    private String direction; // "buy" ili "sell"
}
